### DISCUSSION TITLE
일반톡, 오픈톡 구분방법이 궁금합니다

### DISCUSSION THREAD

**Discussion by wntown** [storycraft/node-kakao#531] [Q&A] [answered]
예제들 통해서 가리기, 채팅발송등은 찾았는데

CLIENT.on('chat', (data, channel) => {
    const sender = data.getSenderInfo(channel);
    if (!sender) return;

    if (data.text === '테스트') {
         channel.sendChat(
             new kakao.ChatBuilder()
                 .append(new kakao.ReplyContent(data.chat))
                 .text('안녕하세요 ')
                 .append(new kakao.MentionContent(sender))
                 .build(kakao.KnownChatType.REPLY));
        channel.hideChat(data.chat); // 가리기
        // channel.sendChat('안녕하세요');
    }
});

이런식으로 찍어봤는데
답장에는 문제가없고
가리기 또한 문제가없지만
일반 채팅방에 전송할경우 에러가 발생하는 이슈가 있는거 같습니다

일반채팅과 오픈채팅을 구분해야 할거같은데 방법이 있을까요?

**Comment by storycraft** [comment, marked as answer]
#320 참고해보세요

**Reply by wntown** [reply]
감사합니다 바로 해결했어요 ㅎㅎ
if (channel.info.type === 'OM') {
    // 이벤트
}

이런식으로도 사용하는 방법은 어떤지 여쭤봐도 될까요?

**Reply by wntown** [reply]
감사합니다 위 부분은 여러가지 방법 테스트 후 해결방법을 찾았습니다
클라이언트 오픈채팅 이벤트만 따로 열어서 실행하는 방법이 가장 깔끔한거같아서 처리했습니다^^

[a comment by an automated account is not shown]
